src/net.cpp

```diff
@@ -1578,30 +1591,30 @@ void CConnman::ThreadDNSAddressSeed()
         LOCK(cs_vNodes);
         int nRelevant = 0;
         for (auto pnode : vNodes) {
-            nRelevant += pnode->fSuccessfullyConnected && ((pnode->nServices & nRelevantServices) == nRelevantServices);
+            nRelevant += pnode->fSuccessfullyConnected && !pnode->fFeeler && !pnode->fOneShot && !pnode->m_manual_connection && !pnode->fInbound;
         }
         if (nRelevant >= 2) {
             LogPrintf("P2P peers available. Skipped DNS seeding.\n");
             return;
         }
     }
 
-    const std::vector<CDNSSeedData> &vSeeds = Params().DNSSeeds();
+    const std::vector<std::string> &vSeeds = Params().DNSSeeds();
     int found = 0;
 
     LogPrintf("Loading addresses from DNS seeds (could take a while)\n");
 
-    for (const CDNSSeedData &seed : vSeeds) {
+    for (const std::string &seed : vSeeds) {
         if (interruptNet) {
             return;
         }
         if (HaveNameProxy()) {
-            AddOneShot(seed.host);
+            AddOneShot(seed);
         } else {
             std::vector<CNetAddr> vIPs;
             std::vector<CAddress> vAdd;
-            ServiceFlags requiredServiceBits = nRelevantServices;
-            std::string host = GetDNSHost(seed, &requiredServiceBits);
+            ServiceFlags requiredServiceBits = GetDesirableServiceFlags(NODE_NONE);
+            std::string host = strprintf("x%x.%s", requiredServiceBits, seed);
```

Comment:
We don't have a prefix for the host.

Reply:
The code is the same as in `Bitcoin Core 0.16`, and is basically refactoring of the existing code.
There are 2 types of DNS nodes:

1) DNS nodes with services, the prefix is used there, there are prefixes from x1. to x31., the prefix is used to provide nodes that support different messages, for normal working in the demon the x9. prefix is required (according to initial check).
2) DNS nodes with no services, the peers are not classified by service, in that case it is not used prefix.

In core 0.16, adding DNS seeds is simplified, so in the code there is automatic check to see which type the DNS node is.
The code look first for DNS nodes type 1, if not found then look (single shot - `AddOneShot`) for DNS nodes type 2.

About qtum, `qtum3.dynu.net` contain no services, the ping doesn't work to `ping x1.qtum3.dynu.net`.

The other DNS addresses actually contain services (probably it is used different script to create them), for example `ping x9.qtum5.dynu.net` do work, the number of IP addresses provided by the services is less then `nslookup qtum5.dynu.net` so probably some update of the DNS nodes to not support services should be made, or update to the needed services with all IPs (x9. for the demon).